sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -475,7 +475,16 @@ private Block BuildTryStatement(TryStatementSyntax tryStatement, Block currentBl
 
             var catchBlocks = tryStatement.Catches
                 .Reverse()
-                .Select(catchClause => BuildBlock(catchClause.Block, CreateBlock(catchSuccessor)))
+                .Select(catchClause =>
+                {
+                    Block catchBlock = BuildBlock(catchClause.Block, CreateBlock(catchSuccessor));
+                    if (catchClause.Filter?.FilterExpression != null)
+                    {
+                        catchBlock = BuildExpression(catchClause.Filter.FilterExpression,

```

Comment:
Does the expression that is built take into account the type of the exception that is being caught when evaluating the expression?

Reply:
This is syntax-based, not types are taken into account here.